### PR TITLE
Webanck demo bron

### PR DESCRIPTION
**Pull request by valentinMachado**
Warning: prior to merging don't forget to change the version number

Warning: alas this PR mixes work on the animation on @webanck contribution on shaders/local-avatar

Realized changes:
* Rename attribute idModel → idRenderData
* Animation new architecture of componenent Render 
* Exemple Avatar Local game
* Re-expose threejs shader related API

### CODE REVIEW

**Comment by webanck** [comment]
@valentinMachado is there anything preventing from accepting this PR ?

File: examples/assets/localScripts/localAvatar.js [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+        5522.95180710312,
+        -3322.608827644959,
+        -110.02345057404978
+      );
+      cube.updateMatrixWorld();
+    });
+  }
+
+  tick() {
+    /*
```
Comment (mathieuLivebardon) [inline examples/assets/localScripts/localAvatar.js:205]: Remove commented code

File: examples/assets/localScripts/localAvatar.js [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+
+    // this.raycaster.ray.origin = pos.add(ref);
+    this.raycaster.ray.origin = origin;
+    this.raycaster.ray.direction = direction;
+    //console.log(origin);
+
+    const intersections = this.raycaster.intersectObjects(buildings, true);
+    if (intersections.length) return intersections[0];
+    return null;
+    //if(!intersections.length) debugger;
```
Comment (mathieuLivebardon) [inline examples/assets/localScripts/localAvatar.js:56]: remove comments

File: examples/assets/localScripts/localAvatar.js [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+    });
+  }
+  buildingsHit(tilesManager, origin, direction) {
+    // const gV = localContext.getGameView();
+
+    // const tilesManager = gV.getLayerManager().tilesManagers;
+    const buildings = [];
+    this.addObjectToArray(buildings, tilesManager, '3d-tiles-layer-building');
+
+    // const pos = this.avatar.getPosition();
```
Comment (mathieuLivebardon) [inline examples/assets/localScripts/localAvatar.js:45]: remove comments

File: examples/assets/localScripts/localAvatar.js [hunk 4]
```diff
@@ -0,0 +1,267 @@
...
+
+    if (!layerManager) throw new Error('no layer called ', layerName);
+
+    layerManager.tiles.forEach(function (t) {
+      const obj = t.getObject3D();
+      if (obj) array.push(obj);
+    });
+  }
+  buildingsHit(tilesManager, origin, direction) {
+    // const gV = localContext.getGameView();
```
Comment (mathieuLivebardon) [inline examples/assets/localScripts/localAvatar.js:39]: remove comments

**Review by mathieuLivebardon — CHANGES_REQUESTED** [review]

**Pull request merged**